src/Clients/CustomResponseClient.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Strictphp\HttpClients\Clients;
+
+use GuzzleHttp\Psr7\Response;
+use Illuminate\Contracts\Filesystem\Filesystem;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Strictphp\HttpClients\Responses\SerializableResponse;
+
+final class CustomResponseClient implements ClientInterface
+{
+    public function __construct(private readonly string $content, private readonly ?Filesystem $filesystem = null)
+    {
+    }
+
+    public function sendRequest(RequestInterface $request): ResponseInterface
+    {
+        if ($this->filesystem !== null && $this->filesystem->exists($this->content)) {
+            $body = (string) $this->filesystem->get($this->content);
+        } elseif (is_file($this->content)) {
+            $body = (string) file_get_contents($this->content);
+        } else {
+            $body = $this->content;
+        }
+
+        if ($body !== '') {
```

Comment:
early return?

Reply:
možnost rozšířit, když by jsi chtěl doplnit převod nějakého content type na konkrétní koncovku, aby jsi to nemusel celé implementovat ale podědil a upravil metodu... já má třídu final?